AK/Function.h

```diff
@@ -254,20 +318,35 @@ class Function<Out(In...)> {
                 "check your capture list if it is a lambda expression, "
                 "and make sure your callable object is not excessively aligned.");
         }
-        VERIFY(m_call_nesting_level == 0);
         using WrapperType = CallableWrapper<Callable>;
+        if (callable_kind == CallableKind::FunctionObject)
+            m_size = sizeof(Callable);
+        else
+            m_size = 0;
+        if constexpr (IsBlockClosure<Callable>) {
+            auto block_size = Block_size(BRIDGE_CAST(void const*, callable));
+            VERIFY(block_size >= Detail::block_layout_size);
+            m_size = block_size - Detail::block_layout_size;
+        }
+
+        VERIFY(m_call_nesting_level == 0);
         if constexpr (alignof(Callable) > inline_alignment || sizeof(WrapperType) > inline_capacity) {
             *bit_cast<CallableWrapperBase**>(&m_storage) = new WrapperType(forward<Callable>(callable));
             m_kind = FunctionKind::Outline;
         } else {
             static_assert(sizeof(WrapperType) <= inline_capacity);
-            new (m_storage) WrapperType(forward<Callable>(callable));
-            m_kind = FunctionKind::Inline;
+            if constexpr (IsBlockClosure<Callable>) {
+                if constexpr (Detail::HaveObjcArc) {
+                    new (m_storage) WrapperType(forward<Callable>(callable));
+                } else {
+                    new (m_storage) WrapperType(reinterpret_cast<Callable>(_Block_copy(forward<Callable>(callable))));
```

Comment:
That forward<> is useless, no?

Reply:
Probably, yes. Especially since cv and ref qualifications for block pointers cause a compile error.

Reply:
`Function& operator=(BlockType&& block)` above must be broken too then, no?

Reply:
I'm not sure what a _block_'s type is, but callable here is a pointer afaik.

Reply:
Hm, the block's type will be `Ret (^)(Args...)`. Which is interpreted as a 'fancy `void const*`' by the compiler. Guess I forgot to add a test for assignment 😅

Reply:
I think I removed any useless forwards. I experimented with telling our trait that references to blocks such as `void(^__strong &)()` are also blocks, but this resulted in some seriously strange compile errors. So we're back to requiring that the compiler strip those off before going anywhere into the Function implementation.